File: dataset/human/java/file_2525.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next();
        String t = sc.next();
        char[] scha = s.toCharArray();
        char[] tcha = t.toCharArray();
        Arrays.sort(scha);
        Arrays.sort(tcha);
        int scnt = scha.length;
        int tcnt = tcha.length;
        boolean minIsS = true;
        if (scnt > tcnt) {
            minIsS = false;
        }
        int ans = -1;
        for (int i = 0; i < Math.min(scnt, tcnt); i++) {
            if (scha[i] == tcha[tcnt - 1 - i])
                continue;
            if (scha[i] < tcha[tcnt - 1 - i]) {
                ans = 1;
                break;
            } else {
                ans = 0;
                break;
            }
        }
        if (ans == -1) {
            if (scnt == tcnt) {
                ans = 0;
            } else if (scnt < tcnt) {
                ans = 1;
            }
        }
        if (ans == 1) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }

    }
}

File: dataset/human/python/file_2525.py
D,G = map(int,input().split())
pc = [list(map(int,input().split())) for i in range(D)]
ans = 10**9
from math import ceil
for i in range(2**D):
    now = 0
    score = 0
    for j in range(D):
        if i>>j & 1:
            now += pc[j][0]
            score += pc[j][0]*100*(j+1) + pc[j][1]

    if score >= G:
        ans = min(ans,now)
    else:
        for j in reversed(range(D)):
            if i>>j & 1 == 0:
                if ceil((G-score)/(100*(j+1))) < pc[j][0]:
                    now += ceil((G-score)/(100*(j+1)))
                    break
        else:
            continue

        if now==0:
            continue
        ans = min(ans,now)

print(ans)
